test/functional/sqlsrv/sqlsrv_ae_output_param_sqltype_binary_size.phpt

```diff
@@ -0,0 +1,279 @@
+            $typeFull = "$dataType($m)";
+        }
+        echo "\nTesting $typeFull:\n";
+            
+        // create and populate table containing binary(m) or varbinary(m) columns
+        $tbname = "test_" . str_replace(array('(', ')'), '', $dataType) . $m;
+        $colMetaArr = array(new AE\ColumnMeta($typeFull, "c_det"), new AE\ColumnMeta($typeFull, "c_rand", null, false));
+        AE\createTable($conn, $tbname, $colMetaArr);
+        $inputs = array(new AE\BindParamOption($inputValues[0], null, "SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_BINARY)"),
+                                new AE\BindParamOption($inputValues[1], null, "SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_BINARY)"));
+        $r;
+        $stmt = AE\insertRow($conn, $tbname, array("c_det" => $inputs[0], "c_rand" => $inputs[1]), $r, AE\INSERT_PREPARE_PARAMS);
+        
+        // create a stored procedure and sql string for calling the stored procedure
+        $spname = 'selectAllColumns';
+        createProc($conn, $spname, "@c_det $typeFull OUTPUT, @c_rand $typeFull OUTPUT", "SELECT @c_det = c_det, @c_rand = c_rand FROM $tbname");
+        $sql = AE\getCallProcSqlPlaceholders($spname, 2);
+        
+        // retrieve by specifying SQLSRV_SQLTYPE_BINARY(n) or SQLSRV_SQLTYPE_VARBINARY(n) as SQLSRV_PARAM_OUT or SQLSRV_PARAM_INOUT
+        foreach($directions as $dir) {
+            echo "Testing as $dir:\n";
+            foreach($sqlTypes as $sqlType) {
+                $maxsqltype = strpos($sqlType, "max");
+                foreach($sqltypeLengths as $n) {
+                    $sqltypeconst;
+                    $sqltypeFull;
+                    if ($maxsqltype !== false) {
+                        $sqltypeconst = SQLSRV_SQLTYPE_VARBINARY('max');
+                        $sqltypeFull = $sqlType;
+                    } else {
+                        $sqltypeconst = call_user_func($sqlType, $n);
+                        $sqltypeFull = "$sqlType($n)";
+                    }
+                    
+                    $c_detOut = '';
+                    $c_randOut = '';
+                    $stmt = sqlsrv_prepare($conn, $sql, array(array(&$c_detOut, constant($dir), SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_BINARY), $sqltypeconst), array(&$c_randOut, constant($dir), SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_BINARY), $sqltypeconst)));
+                    $r = sqlsrv_execute($stmt);
+                    
+                    // check the case when SQLSRV_SQLTYPE length (n) is not the same as the column length (m)
+                    // with AE: should not work
+                    // without AE: should work, except when a SQLSRV_SQLTYPE_VARBINARY length (n) is less than a binary column length (m) for SQLSRV_PARAM_OUT
+                    if (($n != $m || $maxsqltype || $maxcol) && !($maxcol && $maxsqltype)) {
+                        if (AE\isDataEncrypted()) {
+                            if ($r !== false) {
+                            var_dump($n);
+                            var_dump($m);
```

Comment:
is there an indentation issue here or there's a tab here?

Reply:
these var_dumps are for debugging, just forgot to remove them